libr/core/canal.c

```diff
@@ -819,8 +823,20 @@ static int __core_anal_fcn(RCore *core, ut64 at, ut64 from, int reftype, int dep
 		} else if (fcnlen == R_ANAL_RET_END) { /* Function analysis complete */
 			f = r_core_flag_get_by_spaces (core->flags, fcn->addr);
 			if (f && f->name && strncmp (f->name, "sect", 4)) { /* Check if it's already flagged */
-				R_FREE (fcn->name);
-				fcn->name = strdup (f->name);
+				const char *new_name = f->name;
+				if (is_entry_flag (f)) {
+					RListIter *iter;
+					RFlagItem *flag;
+					const RList* flags = r_flag_get_list (core->flags, fcn->addr);
+					r_list_foreach (flags, iter, flag) {
+						if (r_str_startswith (flag->name, "sym.")) {
```

Comment:
Don't check the prefix, you can do something like
```c
if (flag->space && !strcmp (flag->space->name, R_FLAGS_FS_SYMBOLS))
```

Reply:
`entry.*` flags are stored in the `symbols` flag space too :\

Reply:
Hmm ok yeah that sucks. Then it's ok for now I guess.